crates/environ/src/module.rs

```diff
@@ -512,8 +663,8 @@ pub struct Module {
     /// The module "start" function, if present.
     pub start_func: Option<FuncIndex>,
 
-    /// WebAssembly table initializers.
-    pub table_initializers: Vec<TableInitializer>,
+    /// WebAssembly table initialization data, per table.
+    pub table_initialization: PrimaryMap<TableIndex, Option<TableInitialization>>,
```

Comment:
Could this use a style more similar to `MemoryInitialization` to make sure that we're consistent between the two? Where one variant of `TableInitialization` is "list of initializers" and the second variant is "precomputed per-table lists"

Reply:
I tried that at first the but issue is that the mode decision is per-table; so the "FuncTable" mode would then need an inner enum for each table to give the list of indices or the original list of initializers, which just reinvents this enum.

It looks like the `MemoryInitialization` mechanism gets around this by only converting to paged mode if *all* memories are compatible? I'm not sure if we want to do that for tables: it would create a perf cliff where adding one initializer for one imported table slows the whole instantiation down. Open to other ideas though -- thoughts?

Reply:
I think the same argument applies to multi-memory where one out-of-bounds-initializer shouldn't necessarily create a cliff for the other memories. In that sense I'd prefer that we're consistent and fix this later if necessary (either by going the route you've done here or something similar). None of this is really that much of an issue today since modules generally have at most only one funcref table.